keps/sig-node/2172-warn-about-implicitly-insecure-containers/README.md

```diff
@@ -0,0 +1,419 @@
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubernetes does not make it hard enough to do the wrong thing.  Specifically,
+there are things that users do ALL THE TIME that they really should not, such
+as running containers as root.  This KEP aims to make it more obvious that they
+are doing that, and to encourage them to declaring their intent better.
+
+## Motivation
+
+Recent CVEs around containerd come down to containers running as root when they
+shouldn't.  After discussion, I realized many users simply don't know they are
+doing anything wrong.  It's just not obvious.
+
+Many container images do not opt-in to a non-root UID.  Users run these
+containers without specifying a non-root `runAsUser`.  Kubernetes happily runs
+the container as root, whether it needs root or not.
+
+For clarity, we define the following terms:
+
+implicitly-insecure: containers which run as UID or GID 0 and do not set
+`runAsUser` or `runAsGroup` to 0.
+
+explicitly-insecure: containers which run as UID or GID 0 but set
+`runAsUser` or `runAsGroup` to 0.
+
+### Goals
+
+1) To make it more obvious when a pod has implicitly-insecure
+   containers.
+2) To subtly suggest to users that running as root is an error.
+
+### Non-Goals
+
+1) To make it impossible or opt-in to run as root.
+2) To actively impact users who run as root.
+3) To make noise about explicitly-insecure containers.
+
+## Proposal
+
+This proposal includes several parts.
+
+### Report running user and group IDs
+
+This KEP proposes to add 2 fields - `userID` and `groupID` to
+`Pod.status.containerStatuses` to report the UID and GID of the root process
+for each container (as found in `runtimeSpec.process.user`).
```

Comment:
i think this is a good first step. I am guessing people could write opa policies based on these fields easily for the case when the uid and gid are directly coming from the Dockerfile

Reply:
this should be populated regardless of whether its explicit or implicit